beacon_node/beacon_chain/src/light_client_server_cache.rs

```diff
@@ -158,12 +145,20 @@ impl<T: BeaconChainTypes> LightClientServerCache<T> {
             cached_parts.next_sync_committee_branch,
             cached_parts.finality_branch.clone(),
             &attested_block,
-            maybe_finalized_block_root.as_ref(),
+            maybe_finalized_block.as_ref(),
             chain_spec,
         )?;
 
-        let prev_light_client_update =
-            self.get_light_client_update(&store, sync_period, chain_spec)?;
+        let sync_period = block_slot
+            .epoch(T::EthSpec::slots_per_epoch())
+            .sync_committee_period(chain_spec)?;
+
+        // Spec: Full nodes SHOULD provide the best derivable LightClientUpdate (according to is_better_update)
+        // for each sync committee period
+        let prev_light_client_update = match &self.latest_light_client_update.read().clone() {
+            Some(prev_light_client_update) => Some(prev_light_client_update.clone()),
+            None => self.get_light_client_update(&store, sync_period, chain_spec)?,
```

Comment:
I would add the logic to read from `latest_light_client_update` into `get_light_client_update` to ensure consistency. Plus if you read from store once you should write to the `latest_light_client_update` cache

Reply:
[8ef297b](https://github.com/sigp/lighthouse/pull/5915/commits/8ef297b0752031fc8c345decc0d69fe060c88e6e)